pkg/lib/authenticationflow/declarative/intent_check_conflict_and_create_identity.go

```diff
@@ -0,0 +1,110 @@
+package declarative
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/authgear/authgear-server/pkg/api"
+	"github.com/authgear/authgear-server/pkg/api/model"
+	authflow "github.com/authgear/authgear-server/pkg/lib/authenticationflow"
+	"github.com/authgear/authgear-server/pkg/lib/authn/identity"
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/iawaknahc/jsonschema/pkg/jsonpointer"
+)
+
+func init() {
+	authflow.RegisterIntent(&IntentCheckConflictAndCreateIdenity{})
+}
+
+type IntentCheckConflictAndCreateIdenity struct {
+	JSONPointer jsonpointer.T          `json:"json_pointer,omitempty"`
+	UserID      string                 `json:"user_id,omitempty"`
+	Request     *CreateIdentityRequest `json:"request,omitempty"`
+}
+
+var _ authflow.Intent = &IntentCheckConflictAndCreateIdenity{}
+
+func (*IntentCheckConflictAndCreateIdenity) Kind() string {
+	return "IntentCheckConflictAndCreateIdenity"
+}
+
+func (*IntentCheckConflictAndCreateIdenity) CanReactTo(ctx context.Context, deps *authflow.Dependencies, flows authflow.Flows) (authflow.InputSchema, error) {
+	switch len(flows.Nearest.Nodes) {
+	case 0: // next node is NodeDoCreateIdentity, or account linking intent
+		return nil, nil
+	}
+	return nil, authflow.ErrEOF
+}
+
+func (i *IntentCheckConflictAndCreateIdenity) ReactTo(ctx context.Context, deps *authflow.Dependencies, flows authflow.Flows, input authflow.Input) (*authflow.Node, error) {
+	switch len(flows.Nearest.Nodes) {
+	case 0: // next node is NodeDoCreateIdentity, or account linking intent
+		cfg, conflicts, err := i.checkConflictByAccountLinkings(ctx, deps, flows)
+		if err != nil {
+			return nil, err
+		}
+		spec := i.getIdenitySpec()
+		if len(conflicts) == 0 {
+			info, err := newIdentityInfo(deps, i.UserID, spec)
+			if err != nil {
+				return nil, err
+			}
+			return authflow.NewNodeSimple(&NodeDoCreateIdentity{
+				Identity: info,
+			}), nil
+		} else {
+			// Currently skipLogin is always false
+			// We may support always_link_without_login later
+			var skipLogin bool = false
+			var loginFlow string = ""
+			switch cfg.GetAction() {
+			case config.AccountLinkingActionError:
+				spec := spec
+				otherSpec := conflicts[0].ToSpec()
```

Comment:
The current ErrDuplicatedIdentity only contains one identity info.

I am not sure in error case, should we return more than one?

Reply:
Add one more key to return the array of infos